File: 0848. Shifting Letters/shifting_letters.py
# 848. Shifting Letters
# https://leetcode.com/problems/shifting-letters/

from typing import List

def shiftingLetters(s: str, shifts: List[int]) -> str:

  def shift(char: str, amount: int) -> str:
    return chr((ord(char) - 97 + amount) % 26 + 97)

  for i in range(len(shifts) - 2, -1, -1):
    shifts[i] += shifts[i + 1]

  return "".join(shift(*pair) for pair in zip(s, shifts))

File: 0848. Shifting Letters/shifting_letters.ts
const shiftingLetters = (s: string, shifts: number[]): string => {
  for (let i = shifts.length - 2; i > -1; i--){
    shifts[i] += shifts[i + 1];
  }
  return String.fromCharCode(...shifts.map((amount, i) => {
    return (s.charCodeAt(i) - 97 + amount) % 26 + 97
  }));
};
